server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -375,21 +375,25 @@ void removeTombstoneUnderLock(BytesRef uid) {
         }
     }
 
-    private boolean canRemoveTombstone(long currentTime, long pruneInterval, DeleteVersionValue versionValue) {
-        // check if the value is old enough to be removed
-        final boolean isTooOld = currentTime - versionValue.time > pruneInterval;
+    private boolean canRemoveTombstone(long maxTimestampToPrune, long maxSeqNoToPrune, DeleteVersionValue versionValue) {
+        // check if the value is old enough and safe to be removed
+        final boolean isTooOld = versionValue.time < maxTimestampToPrune;
```

Comment:
Why not `<=`?

Reply:
Because we would like to keep delete tombstones for at least one GC cycle.
```diff
-        final boolean isTooOld = currentTime - versionValue.time > pruneInterval;
+        final boolean isTooOld = versionValue.time < maxTimestampToPrune;
```